src/shared/filesystemUtilities.ts

```diff
@@ -115,3 +117,33 @@ export function isInDirectory(d: string, p: string): boolean {
             : value === containedPathPieces[index]
     })
 }
+
+/**
+ * Returns `name.suffix` if it does not already exist in directory `dir`, else appends
+ * a number ("foo-1.txt", "foo-2.txt", etc.).
+ *
+ * To avoid excessive filesystem activity, if all filenames up to `max` exist,
+ * the function instead appends a random string.
+ *
+ * @param dir  Path to a directory
+ * @param name  Filename without extension
+ * @param suffix  Filename suffix, typically an extension (".txt"), may be empty
+ * @param max  Stop searching if all permutations up to this number exist
+ */
+export function getNonexistentFilename(dir: string, name: string, suffix: string, max: number = 999): string {
```

Comment:
Do we ever want to use 999? You're setting the default to 99 everywhere else, does that make more sense as a default?